src/nl/surf/eduhub_rio_mapper/rio.clj

```diff
@@ -161,25 +156,27 @@
        (kenmerken attr-name (attr-name->kenmerk-type attr-name) attr-value)
        [(duoize attr-name) attr-value])]))
 
+(defn wrapper-periodes-cohorten [rio-obj]
+  (fn [key]
+    (rio-obj (if (keyword? key)
+               key
+               (case key
+                 ("aangebodenHOOpleidingsonderdeelPeriode" "aangebodenHOOpleidingPeriode" "aangebodenParticuliereOpleidingPeriode"
+                   "hoOnderwijseenheidPeriode" "hoOpleidingPeriode" "particuliereOpleidingPeriode" "hoOnderwijseenhedenclusterPeriode")
+                 :periodes
+                 ("aangebodenHOOpleidingsonderdeelCohort" "aangebodenHOOpleidingCohort" "aangebodenParticuliereOpleidingCohort")
+                 :cohorten)))))
+
 (defn ->xml [rio-obj object-name]
-  (let [schema (xsd-beheren (name->type object-name))
-        process-attributes (fn [acc kenmerk attr-name]
-                            (into acc (when-let [attr-value (rio-obj (keyword attr-name))]
-                                        (process-attribute attr-name attr-value kenmerk))))
-        process-children (fn [acc type]
-                           (let [{:keys [child-type key]} (converter object-name type)]
-                             (into acc (mapv (fn [child] (->xml child child-type))
-                                             (rio-obj key)))))]
+  {:pre [(string? object-name)]}
+  (let [process-attributes (fn [{:keys [kenmerk name]}]
+                             (when-let [attr-value (rio-obj (keyword name))]
+                               (process-attribute name attr-value kenmerk)))
+        process-children   #(let [child-type (str object-name (if (= (:type %) "AangebodenOpleidingCohort") "Cohort" "Periode"))]
+                             (mapv (fn [child] (->xml child child-type))
+                                   (rio-obj child-type)))]
     (into [(duoize object-name)]
-          (->> schema
-               (reduce (fn [acc {:keys [choice] :as item}]
-                         (if (nil? choice)
-                           (conj acc item)
-                           (into acc choice)))
-                       [])
-               (reduce (fn [acc {:keys [kenmerk name ref type]}]
-                         (if (nil? ref)
-                           (process-attributes acc kenmerk name)
-                           (process-children acc type)))
-                       [])
-               (vec)))))
+          (->> (xsd-beheren (name->type object-name))
+               (mapcat #(get % :choice [%]))                ; choice contains a list, mapcat flattens the list
```

Comment:
bedoel je hier `(mapcat #(get % :choice))` ?

als dat de bedoeling is kan je ook `(mapcat :choice)` doen.